src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -1,26 +1,68 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// eslint-disable-next-line no-use-before-define
 import * as React from 'react';
 
 import BaseChannelItem from './BaseChannelItem';
+import useChatUtilsHook from '../../hooks/core/chat/useChatUtilsHook';
+import useProfileHook from '../../hooks/core/profile/useProfileHook';
 import {BaseChannelItemTypeProps} from '../../../types/component/AnonymousChat/BaseChannelItem.types';
+import {ChannelList} from '../../../types/database/schema/ChannelList.types';
+import {Context} from '../../context';
 import {MessageChannelItemProps} from '../../../types/component/AnonymousChat/MessageChannelItem.types';
 import {calculateTime} from '../../utils/time';
+import {getChatName} from '../../utils/string/StringUtils';
 
 const MessageChannelItem: (props: MessageChannelItemProps) => React.ReactElement = ({
   item,
-  onChannelPressed
+  onChannelPressed,
+  hasFollowButton = false,
+  handleFollow
 }) => {
+  const {signedProfileId} = useProfileHook();
+
+  const determineChatType = (data: ChannelList) => {
+    const isAnonymous = Boolean(data?.channelType === 'ANON_PM');
+    if (!isAnonymous) return BaseChannelItemTypeProps.SIGNED_PM;
+    return BaseChannelItemTypeProps.ANON_PM;
+  };
+
+  const checkIsMe = (data: ChannelList, type: BaseChannelItemTypeProps): boolean => {
+    const rawJson = data?.rawJson;
+    const lastMessage = rawJson?.firstMessage ?? rawJson?.message;
+    const isSystemMessage = lastMessage?.isSystem || lastMessage?.type === 'system';
+
+    if (type === BaseChannelItemTypeProps.SIGNED_PM) {
+      if (isSystemMessage) return false;
+      return lastMessage?.user?.id === signedProfileId;
+    }
+    return item?.user?.isMe ?? false;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkIsMe` function seems to have complex logic and might benefit from some refactoring for readability and maintainability. Consider breaking down the conditions into smaller, well-named functions or variables.

42:
Casting context to any defeats the purpose of TypeScript's static typing. Try to avoid using `any` as much as possible. Instead, define an interface for your context and use that.



<!-- This is an auto-generated comment by OSS CodeRabbit -->